rules/manager.go

```diff
@@ -448,6 +459,32 @@ func (g *Group) run(ctx context.Context) {
 	}
 }
 
+func matches(lbls labels.Labels, matchers ...*labels.Matcher) bool {
+Matcher:
+	for _, m := range matchers {
+		// Skip if matches.
+		if v := lbls.Get(m.Name); m.Matches(v) {
+			continue Matcher
+		}
+		return false
+	}
+	return true
+}
+
+func matchesMatcherSets(matcherSets [][]*labels.Matcher, lbls labels.Labels) bool {
+	if len(matcherSets) == 0 {
+		return true
+	}
+
+	var ok bool
+	for _, matchers := range matcherSets {
+		if matches(lbls, matchers...) {
+			ok = true
```

Comment:
the other query parameters for example 

https://github.com/prometheus/prometheus/pull/12270/files

Reply:
My expectation is that all the Matchers in each `[]*labels.Matcher` will be ANDed, and then the set of those will be `ORed`.  I think this is what we do elsewhere.  
It could be made clearer in comments.

Reply:
Yup, I think this is what we are doing here. `matches` ANDs all the matchers in each set i.e return false if any of them don't match, and then `matchesMatcherSet` takes the set of those results and ORs them i.e returns true if any of them are true. Added a comment for this.